modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -294,15 +295,16 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                 pre_downsampling_size = target.shape[0]
 
                 with torch.autocast(self._device_type, enabled=self._amp):
-                    if self._downsampling_enabled:
+                    if self._downsampling_enabled and not self._sample_before_batch:
```

Comment:
maybe we should have something like self._sampling_mode with an enum with three values: NO_DOWNSAMPLING, SAMPLE_THEN_BATCH, BATCH_THEN_SAMPLE? that would make some statements more readable